psh/dmesg/dmesg.c

```diff
@@ -29,34 +31,86 @@ void psh_dmesginfo(void)
 
 static void psh_dmesg_help(const char *prog)
 {
-	printf("Usage: %s [options] [files]\n", prog);
+	printf("Usage: %s [options]\n", prog);
+	printf("  -D:  disable the printing of messages to the console\n");
+	printf("  -E:  enable the printing of messages to the console\n");
 	printf("  -h:  shows this help message\n");
 }
 
 
+static int psh_kmsgctrl(bool state)
+{
+	int fd = open("devfs/kmsgctrl", O_WRONLY);
+	if (fd < 0) {
+		fd = open("/dev/kmsgctrl", O_WRONLY);
+	}
+
+	if (fd < 0) {
+		perror("dmesg: Failed to open kmsgctrl");
+		return EXIT_FAILURE;
+	}
+
+	size_t err = psh_write(fd, state ? "1" : "0", 2);
+
+	close(fd);
+
+	if (err != 2) {
+		perror("dmesg: Write to kmsgctrl failed");
+		return EXIT_FAILURE;
+	}
+
+	return EXIT_SUCCESS;
+}
+
+
 int psh_dmesg(int argc, char **argv)
 {
 	char buf[256];
```

Comment:
nit: if you're limiting scope either way, maybe move declaration just above second `for`

Reply:
Done